Helper/Hook.php

```diff
@@ -154,4 +168,31 @@ public function verifyWebhook()
             throw new WebapiException(__('Precondition Failed'), 6001, 412);
         }
     }
+
+    /**
+     * Set bugsnag metadata bolt_trace_id
+     */
+    public function setMetadata()
+    {
+        if ($bolt_trace_id = $this->request->getHeader(ConfigHelper::BOLT_TRACE_ID_HEADER)) {
+            $this->bugsnag->registerCallback(function ($report) use ($bolt_trace_id) {
+                $report->setMetaData([
```

Comment:
shouldn't this addMetaData so it won't override existing metadata with store URL?

Reply:
@daisy1754 In fact it is the opposite. setMetaData by default uses array_merge_recursive which keeps multiple values in an array.

```php
    public function setMetaData(array $metaData, $merge = true)
    {
        $this->metaData = $merge ? array_merge_recursive($this->metaData, $metaData) : $metaData;

        return $this;
    }
    public function addMetaData(array $metadata)
    {
        $this->metaData = array_replace_recursive($this->metaData, $metadata);
        $this->metaData = $this->removeNullElements($this->metaData);

        return $this;
    }
```